lib/qo/exceptions.rb

```diff
@@ -8,5 +8,32 @@ module Qo
   # @since 0.2.0
   #
   module Exceptions
+    # Error for not all possible cases being handled. This is optimistic as currently
+    # it will only catch when an actual failure to handle a case is present. This
+    # should be patched in later versions
+    #
+    # @author baweaver
+    # @since 0.99.1
+    class ExhaustiveMatchNotMet < StandardError
+      MESSAGE = 'Exhaustive match required: pattern does not satisfy all possible conditions'
+
+      def initialize
+        super(MESSAGE)
+      end
+    end
+
+    # Not all branches were definied in an exhaustive matcher
+    #
+    # @author baweaver
+    # @since 0.99.1
+    class ExhaustiveMatchMissingBranches < StandardError
```

Comment:
I suggest having common ancestor for all Qo errors and, probably for exhaustive match errors. So end users will be able to catch errors with desired granularity. For example Qo::Error, Qo, MatchError, ExhaustiveMatchNotMet etc.